content/en/blog/_posts/protect-mission-critical-pods-priorityclass/index.md

```diff
@@ -0,0 +1,292 @@
+
+PriorityClass is a cluster-wide API in Kubernetes, and is a part of the `scheduling.k8s.io/v1` API group. It contains a mapping of the PriorityClass name (defined in `.metadata.name`) and an integer value (defined in `.value`). This represents the value that the scheduler uses to determine Pods' relative priority.
+
+Do you think Kubernetes itself uses PriorityClasses? The answer is yes. When you create a cluster using kubeadm, or even if you use a managed Kubernetes service (for example, Azure Kubernetes Service), PriorityClasses are widely used to safeguard the pods which are hosted on the control plane nodes. This aims to ensure that critical cluster components such as CoreDNS and kube-proxy are able to run even if resources are constrained.
+
+Using a special PriorityClass helps to make sure that these pods are up and alive to serve the requests and make sure that the overall cluster is up and running.
+
+```console
+$ kubectl get priorityclass
+NAME                      VALUE        GLOBAL-DEFAULT   AGE
+system-cluster-critical   2000000000   false            82m
+system-node-critical      2000001000   false            82m
+```
+
+The below diagram shows exactly how it works and also contains the details of the example, which will come in the upcoming section (PriorityClass HandsOn Example).
+
+{{< figure src=decision-tree.svg alt="Whenever there is a new incoming pod, this is how kube-scheduler decides their priority and puts them in a queue" title="Decision Tree">}}
+
+### Pod priority and preemption
+
+Pod preemption is a Kubernetes feature that allows the cluster to [preempt](/docs/concepts/scheduling-eviction/pod-priority-preemption/#preemption) lower-priority pods over high-priority pods in the event that there aren't enough resources to run all current Pods. To learn more about this concept, please refer to [Pod Priority and Preemption](/docs/concepts/scheduling-eviction/pod-priority-preemption/) in the Kubernetes documentation.
+
+Pod priority and preemption are Kubernetes features:
+- Which help in deciding a priority while scheduling pods.
+- Which check if the mission-critical pods are up and running and are not evicted.
+- Decide how the cluster will behave in the event of a resource crunch.
+- Decide which lower priority pods are preempted (evicted) from the nodes when there are resource crunch and high priority pods are available in the queue for scheduling.
+
+### PriorityClass requirements
+
+Before you set up PriorityClasses, there are a few things to consider.
+
+1. Decide on which  PriorityClasses are needed.
+   Example: Based on environments, or based on the kind of pods, or based on the kind of applications.
+2. Which PriorityClass resource will be the default for the cluster (otherwise the pods without a `priorityClassName` will be treated as priority 0).
+3. Respective PriorityClass resources in the cluster based on the decision made on #1.
+4. How you'll name your priority classes.
```

Comment:
Will this not be something that's decided as part of #1 when the user decides how the PriorityClass will be implemented within their setup?

Reply:
Reworded the bullet points:
```
1. Decide which PriorityClasses are needed.
For instance, based on environment, type of pods, type of applications, etc.
2. Which PriorityClass resource will be the default for the cluster (otherwise, the pods without a `priorityClassName` will be treated as priority 0).
3. Use a consistent naming convention for all PriorityClasses.
4. Make sure that the pods for your workloads are running with the right PriorityClass.
```